### PR TITLE
Rephrase RDBStorage RuntimeError description

### PR DESCRIPTION
**Pull request by belldandyxtq**
## Motivation
The description was confusing and incorrect

## Description of the changes
This PR rephrases the confusing description of RuntimeError to be moreclear and precise.
The new description is based on discussion with @toshihikoyanase san

### CODE REVIEW

File: optuna/storages/_rdb/storage.py [hunk 1]
```diff
@@ -143,7 +143,8 @@ def objective(trial):
         :exc:`ValueError`:
             If the given `heartbeat_interval` or `grace_period` is not a positive integer.
         :exc:`RuntimeError`:
-            If the a process that was failed by heartbeat but was actually running.
+            When a process tries to finish a trial that has already
+            been set to FAILED by heartbeat.
```
Comment (toshihikoyanase) [inline optuna/storages/_rdb/storage.py:147]: A nit, but how about adding a link to `TrialState.FAIL`?
```suggestion
            been set to :class:`~optuna.trial.TrialState.FAIL` by heartbeat.
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. The change seems reasonable to me since the new sentence is less ambiguous than the current one.

**Comment by belldandyxtq** [comment]
Thank you for your suggestion, I have fixed it

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!
image

**Review by hvy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**